src/xdp-utils.c

```diff
@@ -563,14 +563,65 @@ xdp_has_path_prefix (const char *str,
     }
 }
 
-void
-cleanup_temp_file (void *p)
+static gboolean
+ensure_sealed_memfd (int      fd,
+                     GError **error) {
+  int saved_errno;
+
+  if (fcntl(fd, F_ADD_SEALS, F_SEAL_GROW | F_SEAL_SHRINK | F_SEAL_WRITE) == -1)
+    {
+      saved_errno = errno;
+
+      g_set_error (error,
+                   G_IO_ERROR,
+                   g_io_error_from_errno (saved_errno),
+                   "Impossible to seal the given file descriptor"
+                   "(fcntl F_ADD_SEALS: %s)", g_strerror (saved_errno));
+      return FALSE;
+    }
+
+  return TRUE;
+}
+
+static int
+bytes_to_fd (GBytes *bytes)
```

Comment:
The name itself is useless. A fd can represent anything but this specifically creates a memfd.